### Right-Tailed - One-Sample Z-Test for Proportions

**1 - What is it?**

A `right-tailed one-sample Z-test for proportions` checks whether a `sample proportion` is significantly higher than a `known population (benchmark) proportion`. 

**Use this when::** 
- Outcome is binary (yes/no, convert/not convert)
- You are comparing `one sample vs fixed benchmark`
- Sample size is large
- You are testing for an `increase only`

**Think of it like:** "Is my observed conversion rate really higher than the expected baseline, or is this just random variation?"

**2 - Real-Life Example: Right-Tailed One-Sample Z-Test for Proportion**

`Scenario:`

Your marketing team claims:
"This campaign's conversion rate is higher than our usual 3% benchmark."

So we collect data:
- Total visitors = 1,000
- Conversions = 45
- Observed conversion rate = 4.5%
- Benchmark (population proportion) = 3%

`Question:`

**Is 4.5% significantly higher than 3%**

This is a right-tailed one-sample Z-test for proportions.

**3 - Hypotheses**

- H0 (Null): p = p0 = 0.03 (No change)
- H1 (Alternate): p > 0.03 (There is a increase)

This is *right-tailed* because we care only about an increease.

**4 - Formula & Parameters**

Z = (Sample proportion - Population proportion)/Standard Error

Standard Error (SE) = SQRT((Population proportion * (1-Population proportion))/n)

**Parameters:**
- Conversion x : 45
- Sample size n: 1000
- Benchmark p0: 0.03
- Significance level(alpha): 0.05
- Confidence Interval: 95%

Here, **population std is known and sample size>=30**, So we can do z-test, specifically *right-tailed one-sample z-test for proportion.*

**5 - Manual Calculation (Step-by-Step)**

- Step1: Sample Proportion

    p^ (p-hat) = 45/1000 = 0.045

- Step2: Standard Error

    SE = SQRT(0.03*(1-0.03)/1000) = SQRT(0.00000291) = 0.00539

- Step3: Z-Statistic

    Z = (0.045 - 0.03) / 0.00539 = 2.78

- Step4: P-Value (right-tailed)

    p = 1 - stats.norm.cdf(2.78) = 0.00027

**6 - Python Implementation (Step-by-Step)**

In [1]:
import numpy as np
from scipy import stats

# ------------------------------------------------------------
# Step 1: Inputs
# ------------------------------------------------------------

x = 45     #number of conversions
n = 1000    #total visitors
p0 = 0.03   #benchmark proportion
alpha = 0.05  #5% significance level (right-tailed)

# ------------------------------------------------------------
# Step 2: Sample proportion
# ------------------------------------------------------------
p_hat = x/n
print(f"Sample Proportion (p_hat): {p_hat:.4f}")

# ------------------------------------------------------------
# Step 3: Standard Error (use p0!)
# ------------------------------------------------------------
se = np.sqrt((p0 * (1-p0))/n)
print(f"Standard Error (SE): {se:.6f}")

# ------------------------------------------------------------
# Step 4: Z-stat
# ------------------------------------------------------------
z_stat = (p_hat - p0)/se
print(f"Z-Statistics: {z_stat:.4f}")

# ------------------------------------------------------------
# Step 5: P-Value (Right-Tailed) 
# ------------------------------------------------------------
p_value = 1 - stats.norm.cdf(z_stat)
print(f"P-Value: {p_value:.4f}")

# ------------------------------------------------------------
# Step 6: Critical Values (Right-Tailed)
# ------------------------------------------------------------
critical_value = stats.norm.ppf(1-alpha)
print(f"Critical Values: {critical_value:.4f}")
      
# ------------------------------------------------------------
# Step 7: Decision Based on P-Value
# ------------------------------------------------------------
print("\n----------- Decision Based on P-Value ---------------")
if p_value < alpha:
    print("Reject H0: Conversion rate is significantly higher than benchmark.")
else:
    print("Fail to Reject H0: No significant increase detected.")

print("\n----------- Decision Based on Z vs Critical Values ------------")
if z_stat > critical_value:
    print("Reject H0: Statistically significant differences detected.")
else:
    print("Fail to Reject H0: No statistically meaningful differences.")

Sample Proportion (p_hat): 0.0450
Standard Error (SE): 0.005394
Z-Statistics: 2.7806
P-Value: 0.0027
Critical Values: 1.6449

----------- Decision Based on P-Value ---------------
Reject H0: Conversion rate is significantly higher than benchmark.

----------- Decision Based on Z vs Critical Values ------------
Reject H0: Statistically significant differences detected.


#### 7 - Business Interpretation
- Observed conversion rate = 4.5%
- Benchmark = 3%
- Z = 2.78 -> 2.78 standard errors above expectation
- p = 0.27% -> extremely unlikely due to chance

**Final Conclusion:**
The campaign's conversion rate is statistically significantly higher than the benchmark. The improvement is real, not random noise.